File: Data Structures & Algorithms/daily-temperatures/submission-0.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        let stack = []
        let ans = Array.from(temperatures.length).fill(0)

        for(let i = temperatures.length - 1; i>=0; i--){
            while(stack.length && temperatures[stack.at(-1)] <= temperatures[i]){
                stack.pop()
            }
            if(stack.length){
                ans[i] = stack.at(-1) - i
            }
            else{
                ans[i] = 0
            }
            stack.push(i)
        } 
        return ans
    }
}

File: Data Structures & Algorithms/baseball-game/submission-0.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:

        stack = []

        for op in operations:

            if op == "C":
                stack.pop()

            elif op == "D":
                res = stack[-1] * 2
                stack.append(res)

            elif op == "+":
                res = stack[-1] + stack[-2]
                stack.append(res)

            else:
                stack.append(int(op))

        return sum(stack)